### DISCUSSION TITLE
Is it possible to directly create a D-Foundations Shallow Foundations model

### DISCUSSION THREAD

**Discussion by RutgerTAUW** [Deltares/GEOLib#202] [Q&A]
I want to use geolib to create a D-Foundations Shallow Foundations model. However when I look in the code it seems that I can only create a BearingPilesModel or a TensionPilesModel. There seems to be a workaround to create a TensionPilesModel and then post-process the .foi file manually but it would be ideal to create a Shallow Foundations model directly. Is this possible?

**Comment by zantDelta** [comment]
GEOLib was only implemented for Bearing piles and Tension piles. If anyone would like to add functionality for Shallow foundations, we would be happy to review it.
